### PR TITLE
init cli

### PR DESCRIPTION
**Pull request by pixelsoccupied**
boiler plate code for cli

### CODE REVIEW

File: cmd/compare.go [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+	Use:   "compare",
+	Short: "Compare two sets of k8s resources",
+	Long:  `Compare two sets of k8s resources using two directory paths`,
+	Args: func(cmd *cobra.Command, args []string) error {
+		// Run the custom validation logic
+		if util.IsDirectory(ResourceDir) && util.IsDirectory(ResourceDir) {
+			return nil
+		}
+
+		return fmt.Errorf("both path must be a directory")
```
Comment (imiller0) [inline cmd/compare.go]: trivial...
```suggestion
		return fmt.Errorf("both paths must be a directory")
```

File: cmd/compare.go [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	},
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("compare called")
+	},
+}
+
+func init() {
+	rootCmd.AddCommand(compareCmd)
+
+	compareCmd.Flags().StringVarP(&ReferenceDir, "reference", "", "", "Reference directory such as source-cr directory from ZT")
```
Comment (imiller0) [inline cmd/compare.go]: minor
```suggestion
	compareCmd.Flags().StringVarP(&ReferenceDir, "reference", "", "", "Reference configuration directory such as source-cr directory from ZT")
```

File: cmd/compare.go [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	compareCmd.Flags().StringVarP(&ResourceDir, "resource", "", "", "Resource directory to read from")
```
Comment (imiller0) [inline cmd/compare.go]:
```suggestion
	compareCmd.Flags().StringVarP(&ResourceDir, "resource", "", "", "User configuration directory to read from")
```

File: cmd/version.go [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+	"github.com/spf13/cobra"
+)
+
+// versionCmd shows the current version
+var versionCmd = &cobra.Command{
+	Use:   "version",
+	Short: "Reference-validator version",
+	Long:  `Reference-validator version`,
+	Run: func(cmd *cobra.Command, args []string) {
+		fmt.Println("reference-validator is currently a work-in-progress and is intended for proof-of-concept/demonstration only")
```
Comment (imiller0) [inline cmd/version.go]: Maybe give it a 0.0.1 or something?

Comment (pixelsoccupied) [reply cmd/version.go]: update to return 0.0.1

```
go run main.go version                   
0.0.1
```

**Review by imiller0 — APPROVED** [review]
/lgtm

**Pull request merged**